### ISSUE TITLE
dolt crashes when connecting with DBeaver

### ISSUE THREAD

**Issue by DeflateAwning** [dolthub/dolt#3632]
## How to Reproduce
1. `dolt clone dolthub/museum-collections`
2. `dolt sql-server`
3. In DBeaver: Connect with MySQL or MariaDB. Expand Databases, then expand Tables, then double-click on the `objects` table.

## Result
The following crash error comes from the `dolt sql-server` process.
```
xxx@localhost:~/dolt$ dolt sql-server
Starting server with Config HP="localhost:3306"|T="28800000"|R="false"|L="info"
2022-06-16T18:55:14Z INFO [conn 1] NewConnection {DisableClientMultiStatements=false}
2022-06-16T18:55:14Z INFO [conn 2] NewConnection {DisableClientMultiStatements=false}
2022-06-16T18:55:15Z WARN [conn 2] error running query {connectTime=2022-06-16T18:55:15Z, connectionDb=, error=unsupported feature: SHOW PLUGINS, query=SHOW PLUGINS}
panic: unexpected type

goroutine 203 [running]:
github.com/dolthub/go-mysql-server/sql.mustUint64(...)
        /go/pkg/mod/github.com/dolthub/go-mysql-server@v0.11.1-0.20220610214256-7b23dffe828c/sql/numbertype.go:1008
github.com/dolthub/go-mysql-server/sql.numberTypeImpl.SQL({0x492ea0?}, {0x0?, 0x0, 0x0?}, {0x1ab57c0?, 0x2fee830?})
        /go/pkg/mod/github.com/dolthub/go-mysql-server@v0.11.1-0.20220610214256-7b23dffe828c/sql/numbertype.go:359 +0x518
github.com/dolthub/go-mysql-server/server.rowToSQL({0xc000b34d10, 0x16, 0x0?}, {0xc000ce29a0, 0x16, 0x1?})
        /go/pkg/mod/github.com/dolthub/go-mysql-server@v0.11.1-0.20220610214256-7b23dffe828c/server/handler.go:720 +0x115
github.com/dolthub/go-mysql-server/server.(*Handler).doQuery.func4()
        /go/pkg/mod/github.com/dolthub/go-mysql-server@v0.11.1-0.20220610214256-7b23dffe828c/server/handler.go:500 +0x65c
golang.org/x/sync/errgroup.(*Group).Go.func1()
        /go/pkg/mod/golang.org/x/sync@v0.0.0-20210220032951-036812b2e83c/errgroup/errgroup.go:57 +0x67
created by golang.org/x/sync/errgroup.(*Group).Go
        /go/pkg/mod/golang.org/x/sync@v0.0.0-20210220032951-036812b2e83c/errgroup/errgroup.go:54 +0x8d
```

## Environment
* Tested on Windows 10 and Ubuntu 20
* dolt version 0.40.9
* DBeaver v22.1.0
* Repo: https://www.dolthub.com/repositories/dolthub/museum-collections

**Comment by andrew-wm-arthur** [comment]
@DeflateAwning thanks for the repro, we'll dig in

**Comment by bpf120** [comment]
Hi @DeflateAwning , thanks for filing! Are you using Dolt for bounties or do you have another use case?

Swing by our Discord if you need anything. 

https://discord.com/invite/RFwfYpu

**Comment by andrew-wm-arthur** [comment]
@DeflateAwning I've repro'd this bug in `0.40.9`. It appears for be fixed in Dolt `main`, but I've added a patch to our SQL engine to prevent this crash in the future. If you'd like, you can build Dolt from `main`, otherwise I'll make a release later today with the patch.

**Comment by DeflateAwning** [comment]
Fixed in the v0.40.10 release! Thanks for doing that so quickly!
